expapi/v2/metrics_handlers.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v2
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"net/http"
+	"time"
+
+	restful "github.com/emicklei/go-restful"
+	"github.com/golang/glog"
+
+	"k8s.io/heapster/expapi/v2/types"
+	model_api "k8s.io/heapster/model"
+	"k8s.io/kubernetes/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+)
+
+// errModelNotActivated is the error that is returned when manager.cluster
+// has not beed initialized.
+var errModelNotActivated = errors.New("the model is not activated")
+
+// RegisterMetrics registers the Metrics API endpoints.
+// All endpoints that end with a {metric-name} also receive a start time query parameter.
+// The start and end times should be specified as a string, formatted according to RFC 3339.
+// These apis are experimental, so they may change or disappear in the future.
+func (a *Api) RegisterMetrics(container *restful.Container) {
+	ws := new(restful.WebService)
+	ws.
+		Path("/experimental/v2").
```

Comment:
This isn't an endpoint - it's the path for the whole WebService. Basically, the prefix and other common parameters for the endpoints under it. So `/nodeMetrics` below becomes `/experimental/v2/nodeMetrics`, etcetera.